### ISSUE TITLE
Report to invert 'negative' condition when directly returning values in all decision branches

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#597]
We should report in case we have "negative" `if` statement or a conditional statements, and in all decision branches we simple return values.

Following code should trigger a warning:
```c#
return !int.TryParse(text, out value) ? null : value;
```
or:
```c#
if (!int.TryParse(text, out value))
{
    return null;
}

return value;
```
In addition, warnings should be reported for `xyz is false` as in :
```c#
return int.TryParse(text, out value) is false ? null : value;
```
or:
```c#
if (int.TryParse(text, out value) is false)
{
    return null;
}

return value;
```

In contrast to the examples above, following code should NOT trigger a warning because it contains a positive condition.

```c#
return int.TryParse(text, out value) ? value : null;
```

Same for following code:
```c#
if (int.TryParse(text, out value))
{
    return value;
}

return null;
```

**Comment by RalfKoban** [comment]
Same applies when throwing exceptions, as in following code (which also should trigger a warning):
```c#
if (!something.Aborted)
{
    return;
}

throw new OperationCanceledException();
```
